src/utils/constants/styles.ts

```diff
@@ -6,17 +6,47 @@ const CLASS_NAMES = Object.freeze({
   DARK: 'dark'
 });
 
-const LIGHT_DARK_BORDER_CLASSES = clsx(
-  // TODO: could be reused
+// MEMO: inspired by https://mui.com/components/buttons/
+const BORDER_CLASSES = clsx(
   'border',
-  // MEMO: inspired by https://mui.com/components/buttons/
   'border-black',
   'border-opacity-25',
   'dark:border-white',
   'dark:border-opacity-25'
 );
+const DISABLED_BORDER_CLASSES = clsx(
+  'border',
+  'border-black',
+  'border-opacity-10',
+  'dark:border-white',
+  'dark:border-opacity-10'
+);
+
+// MEMO: inspired by https://mui.com/customization/dark-mode/#dark-mode-with-custom-palette
+const DISABLED_BACKGROUND_CLASSES = clsx(
+  'bg-black',
+  'bg-opacity-10',
+  'dark:bg-white',
+  'dark:bg-opacity-10'
+);
+const DISABLED_TEXT_CLASSES = clsx(
+  'text-black',
+  'text-opacity-25',
+  'dark:text-white',
+  'dark:text-opacity-30'
+);
+
+const TEXT_CLASSES = clsx(
+  'text-black',
+  'text-opacity-90',
+  'dark:text-white'
+);
 
 export {
   CLASS_NAMES,
-  LIGHT_DARK_BORDER_CLASSES
+  BORDER_CLASSES,
+  DISABLED_BORDER_CLASSES,
+  DISABLED_BACKGROUND_CLASSES,
+  DISABLED_TEXT_CLASSES,
+  TEXT_CLASSES
```

Comment:
I configured these styling constants like so because they were duplicated across the project. I picked up the underlying classes from Material UI e.g. border styling in light/dark modes when enabled and disabled so I think the practices are reliable.
Even if we move over to another styling library other than tailwindcss, we can continuously use them by porting them since the underlying practices are reliable and reusable. In that consideration, I think it's worth having these constants in a good place.